include/utils.hxx

```diff
@@ -222,10 +223,11 @@ public:
   void ensureUnique() {
     data.ensureUnique();
   }
-  
+
+  /// Underlying 1D storage array
+  Array<T> data;
```

Comment:
This looks like it could be dangerous, although I can see several advantages to doing this. 

Would making Fields a friend of these classes be one way to avoid exposing this more broadly?

The "risk" might be worth it considering some of the benefits.

Reply:
Agree this could be dangerous. Would having a function `getData` to return the Array be better? That would allow Fields to be constructed from Matrix/Tensor, without giving too much access to the internals. It does still rely on implementation detail though.

For reading in grid files, I'd like to be able to select a region of a `Tensor` and copy it into a `Field3D`, based on a global offset. I don't think indexing can use `Ind3D` since the array size may be larger than a field and a different shape. Is there an efficient way to convert from `Ind3D` to (x,y,z), add an offset to each index, then use that to index a Tensor?

Reply:
Another way to do it would be something like:

~~~~
Region<Ind3D> theTensorSubset(xstartSection, xendSection......);
auto tensorIndices = theTensorSubset.getIndices();
auto fieldIndices = field.getRegion("RGN_ALL").getIndices();
ASSERT1(tensorIndices.size() == fieldIndices.size();
int count = 0;
BOUT_FOR_SERIAL(i, field.getRegion("RGN_ALL")) {
theField[i] = theTensor[tensorIndices[count]];
count++;
}
~~~~

This doesn't use a